src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -1366,7 +1367,7 @@ impl EnumMemberDescriptionFactory<'ll, 'tcx> {
                             let value = (i.as_u32() as u128)
                                 .wrapping_sub(niche_variants.start().as_u32() as u128)
                                 .wrapping_add(niche_start);
-                            let value = value & ((1u128 << niche.value.size(cx).bits()) - 1);
+                            let value = truncate(value, niche.value.size(cx));
```

Comment:
Oh wow I don't know what I thought when I wrote that code, maybe I just forgot the semantics of `<<`, even when it doesn't panic - at least it's not UB!

*Nevermind*, it wasn't me, see https://github.com/rust-lang/rust/pull/55701#discussion_r270227197.
Also, the `assert_eq!` I suggested in https://github.com/rust-lang/rust/commit/da7b6b4b4dcdf9adf25eca12293be029ce9ce24c, during #54004, was removed, which means debuggers may be getting bad values now (if we started allowing `u128` `enum` discriminants).